In [1]:
import pandas as pd

In [2]:
path = "../data/CMOS/32nm/NOT/simu_0/treated_data.txt"

df = pd.read_csv(path, index_col=0)

Renaming the column 'tensao' to 'voltage' to maintain the English standard

In [3]:
df.rename(columns={'tensao' : 'voltage'})

,nmos@var,pmos@varp,tphl,tplh,iint,temper,voltage,width_pmos,width_nmos,length_n&p
0,0.5166,-0.4341,9.607000e-12,8.368000e-12,-1.330000e-15,-25.0,0.6,1.400000e-07,7.000000e-08,3.200000e-08
1,0.5064,-0.4523,9.424000e-12,8.791000e-12,-1.337000e-15,-25.0,0.6,1.400000e-07,7.000000e-08,3.200000e-08
2,0.4909,-0.4618,9.156000e-12,9.046000e-12,-1.388000e-15,-25.0,0.6,1.400000e-07,7.000000e-08,3.200000e-08
3,0.4882,-0.4667,9.111000e-12,9.161000e-12,-1.395000e-15,-25.0,0.6,1.400000e-07,7.000000e-08,3.200000e-08
4,0.5275,-0.4342,9.809000e-12,8.899000e-12,-1.351000e-15,-25.0,0.6,1.400000e-07,7.000000e-08,3.200000e-08
...,...,...,...,...,...,...,...,...,...,...
167995,0.5126,-0.4225,9.682000e-12,2.399000e-11,-1.790000e-15,50.0,0.9,7.000000e-08,7.000000e-08,3.200000e-08
167996,0.4847,-0.4406,9.130000e-12,2.513000e-11,-2.135000e-15,50.0,0.9,7.000000e-08,7.000000e-08,3.200000e-08
167997,0.4997,-0.4561,9.415000e-12,2.639000e-11,-1.833000e-15,50.0,0.9,7.000000e-08,7.000000e-08,3.200000e-08
167998,0.5154,-0.4417,9.738000e-12,2.516000e-11,-1.677000e-15,50.0,0.9,7.000000e-08,7.000000e-08,3.200000e-08


Inputs (7 total)
x = [nmos@var, pmos@var, temper, voltage, width_nmos, width_pmos, length]

Outputs (3 total)
y = [tphl, tplh, iint]

Variables (10 total)

- **nmos@var**: variação de processo do transistor NMOS
- **pmos@var**: variação de processo do transistor PMOS
- **tphl**:	tempo de propagação da saída High -> Low
- **tplh**: tempo de propagação da saída Low -> High
- **iint**:	energia dissipada pelo circuito
- **temper**:	temperatura de operação do circuito
- **voltage**: tensão de alimentação (vdd)
- **width_pmos**:	largura W do transistor PMOS
- **width_nmos**: largura W do transistor NMOS
- **length_n&p**: comprimento do canal L dos transistores NMOS e PMOS

Checking if there're null values and how many different values in each variable

df.info()

In [4]:
df.index

Index([     0,      1,      2,      3,      4,      5,      6,      7,      8,
            9,
       ...
       167990, 167991, 167992, 167993, 167994, 167995, 167996, 167997, 167998,
       167999],
      dtype='int64', length=155077)

The difference between the length (155.077) and the range of indexes values (0 - 167.999) means treated_data really has been treated. There're gaps that have been taken out of the dataset.

In [5]:
df.size

1550770

In [6]:
df.shape

(155077, 10)

In [7]:
df.nunique()

nmos@var        513
pmos@varp       489
tphl           8200
tplh          10430
iint          14040
temper            6
tensao            4
width_pmos        5
width_nmos        2
length_n&p        3
dtype: int64

There aren't fixed variables values.

- **nmos@var**: Existem 513 valores diferentes
- **pmos@var**:	Existem 489 valores diferentes
- **tphl**:	Existem 8200 valores diferentes
- **tplh**:	Existem 10430 valores diferentes
- **iint**:	Existem 14040 valores diferentes
- **temper**:	Existem 6 temperaturas diferentes
- **voltage**:	Existem 4 tensões diferentes
- **width_pmos**:	Existem 5 larguras diferentes
- **width_nmos**:	Existem 2 larguras diferentes
- **length_n&p**: Existem 3 valores diferentes

In [17]:
df.describe(include="all")

,nmos@var,pmos@varp,tphl,tplh,iint,temper,tensao,width_pmos,width_nmos,length_n&p
count,155077.000000,155077.000000,1.550770e+05,1.550770e+05,1.550770e+05,155077.000000,155077.000000,1.550770e+05,1.550770e+05,1.550770e+05
mean,0.509108,-0.450684,6.592126e-11,-6.440956e-10,-2.063419e-03,37.467033,0.750069,2.266666e-07,8.083333e-08,3.238095e-08
std,0.016825,0.015274,7.263116e-10,2.477490e-09,2.024013e-01,42.982289,0.111693,1.234463e-07,2.531751e-08,4.477225e-09
min,0.461300,-0.493300,-3.819000e-12,-1.000000e-08,-3.342370e+01,-25.000000,0.600000,7.000000e-08,7.000000e-08,2.000000e-08
25%,0.497800,-0.461900,9.825000e-12,7.604000e-12,-2.351000e-15,0.000000,0.700000,1.400000e-07,7.000000e-08,3.200000e-08
50%,0.509000,-0.450500,1.226000e-11,1.080000e-11,-1.657000e-15,50.000000,0.800000,1.400000e-07,7.000000e-08,3.200000e-08
75%,0.520200,-0.440300,1.476000e-11,1.874000e-11,-1.336000e-15,75.000000,0.800000,3.500000e-07,7.000000e-08,3.200000e-08
max,0.586100,-0.405800,1.018000e-08,2.631000e-10,2.933300e+01,100.000000,0.900000,4.200000e-07,1.400000e-07,4.000000e-08


In [20]:
df.isnull().sum()

nmos@var      0
pmos@varp     0
tphl          0
tplh          0
iint          0
temper        0
tensao        0
width_pmos    0
width_nmos    0
length_n&p    0
dtype: int64

As we've seen before, there is no null values

The dataset is composed only of quantitative (numerical) variables, there's none categorical (non-numerical) variables

In [24]:
df.value_counts()

nmos@var  pmos@varp  tphl          tplh          iint           temper  tensao  width_pmos    width_nmos    length_n&p  
0.5097    -0.4458    1.118000e-11  5.535000e-12  -1.569000e-15  -25.0   0.6     2.800000e-07  7.000000e-08  3.200000e-08    2
                     1.162000e-11  6.258000e-12  -1.802000e-15   0.0    0.6     2.800000e-07  7.000000e-08  3.200000e-08    2
                     1.216000e-11  7.174000e-12  -2.086000e-15   25.0   0.6     2.800000e-07  7.000000e-08  3.200000e-08    2
                     1.006000e-11  1.159000e-11  -1.726000e-15   25.0   0.6     1.400000e-07  7.000000e-08  3.200000e-08    2
                     1.370000e-11  7.213000e-12  -2.729000e-15   50.0   0.6     3.500000e-07  7.000000e-08  3.200000e-08    2
                                                                                                                           ..
0.5126    -0.4225    9.682000e-12  2.399000e-11  -1.790000e-15   50.0   0.9     7.000000e-08  7.000000e-08  3.200000e-08   

In [26]:
df.corr()

,nmos@var,pmos@varp,tphl,tplh,iint,temper,tensao,width_pmos,width_nmos,length_n&p
nmos@var,1.000000,0.036060,-0.021852,-0.023326,-0.006767,0.004609,-0.000900,-0.003142,-0.001915,-0.012376
pmos@varp,0.036060,1.000000,-0.001588,-0.001887,0.001435,0.000603,-0.000204,-0.000203,-0.000124,-0.000800
tphl,-0.021852,-0.001588,1.000000,0.018602,-0.000556,0.017631,0.012596,-0.051841,-0.033642,-0.201967
tplh,-0.023326,-0.001887,0.018602,1.000000,0.037966,0.009129,0.001332,0.184180,0.112636,0.736120
iint,-0.006767,0.001435,-0.000556,0.037966,1.000000,-0.006553,-0.000047,0.007157,0.004362,0.028192
temper,0.004609,0.000603,0.017631,0.009129,-0.006553,1.000000,0.002135,0.000538,0.000328,0.002121
tensao,-0.000900,-0.000204,0.012596,0.001332,-0.000047,0.002135,1.000000,-0.000432,-0.000263,-0.001700
width_pmos,-0.003142,-0.000203,-0.051841,0.184180,0.007157,0.000538,-0.000432,1.000000,0.670151,-0.059735
width_nmos,-0.001915,-0.000124,-0.033642,0.112636,0.004362,0.000328,-0.000263,0.670151,1.000000,-0.036408
length_n&p,-0.012376,-0.000800,-0.201967,0.736120,0.028192,0.002121,-0.001700,-0.059735,-0.036408,1.000000
